torch/csrc/jit/python/script_init.cpp

```diff
@@ -866,6 +866,39 @@ void initJitScriptBindings(PyObject* module) {
                 // Similar to Tensor's `__hash__`, which is `id()`.
                 return std::hash<c10::ivalue::Object*>{}(self._ivalue().get());
               })
+          .def(
+              "_clone",
```

Comment:
Why does this need to be implemented in C++? We could have a deepcopy(torchbind_obj) in Python that literally calls unflatten/flatten

Reply:
I'm not sure how to attach a __deepcopy__ implementation to the torchbind python object, so I implemented it here. (updated it to directly implement __deepcopy__ instead of _clone)

Reply:
What I meant was we could have a "torch.utils.torchbind.deepcopy(obj)" (or something) that we invoke in AOTAutograd to do the copy. Though it's not clear to me which is better.

The other things I'm not sure about are:
- Is it guaranteed that flatten + unflatten on a real torchbind object gets us the correct behavior? This might be true for our test cases, but I'm not sure if we require this to be true in general

Reply:
> What I meant was we could have a "torch.utils.torchbind.deepcopy(obj)" (or something) that we invoke in AOTAutograd to do the copy. Though it's not clear to me which is better.

I was thinking maybe implementing `__deepcopy__` might be better as we can also use it to copy the inputs and run with the first custom op pass (example in D64124825).

> Is it guaranteed that flatten + unflatten on a real torchbind object gets us the correct behavior? This might be true for our test cases, but I'm not sure if we require this to be true in general.

Yeah I guess it might be the case that the user's flatten/unflatten calls are not actually copying... for example the original tensor queue flatten implementation did not do a copy of the queue's contents. Is there a way to require this to be true? 

> Real tensor prop implies that we shouldn't need any user intervention. Requiring the user to define unflatten/flatten is user intervention. Is there anything else we can do here?

User intervention is only necessary if the operation results in a mutation on the obj, which I guess is somewhat common since the goal of the torchbind obj is to store state.. I'm not sure what's another thing we could do to preserve the mutations...

Reply:
Updated to just implement a `__deepcopy__`